### PR TITLE
improve type for control

### PR DESCRIPTION
**Pull request by kotarella1110**
Improved type for useForm's `control` object.
This fix improves DX for developers like us who contribute to RHF.

### CODE REVIEW

File: src/useFieldArray.ts [hunk 1]
```diff
@@ -37,7 +38,7 @@ export function useFieldArray<
 
     for (const key in fieldsRef.current) {
       if (isMatchFieldArrayName(key, name)) {
-        unregister(key, true);
+        unregister(key);
```
Comment (kotarella1110) [inline src/useFieldArray.ts]: Again, the second argument `true` doesn't seem to make sense.

```typescript
unregister(name: FieldName<FormValues>): void;
unregister(names: FieldName<FormValues>[]): void;
unregister(names: FieldName<FormValues> | FieldName<FormValues>[]): void;
```

File: src/controller.tsx [hunk 2]
```diff
@@ -107,6 +107,7 @@ const Controller = ({
     () => {
       registerField();
       return () => {
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (bluebill1049) [inline src/controller.tsx]: Wrong spot?

File: src/useFieldArray.ts (hunk 1 above)
Comment (bluebill1049) [reply src/useFieldArray.ts]: Oops mistake

**Comment by bluebill1049** [comment]
Amazing!!! 🤗❤️😉

**Review by bluebill1049 — DISMISSED** [review]

File: src/controller.tsx (hunk 2 above)
Comment (bluebill1049) [inline src/controller.tsx]:
```suggestion
```

Comment (kotarella1110) [reply src/controller.tsx]: its correct!

スクリーンショット 2020-01-17 9 06 30

Comment (bluebill1049) [reply src/controller.tsx]: hoist it up as a variable, it should resolve it.

File: src/types.ts [hunk 3]
```diff
@@ -172,7 +172,58 @@ export type FormValuesFromErrors<Errors> = Errors extends FieldErrors<
...
+    Record<string, (values: any) => void>
+  >;
+  fieldArrayNamesRef: React.MutableRefObject<Set<string>>;
+  isDirtyRef: React.MutableRefObject<boolean>;
+  defaultValuesRef: React.MutableRefObject<
+    DeepPartial<FormValues> | FormValues[FieldName<FormValues>]
+  >;
+};
+
+export type ControllerProps<ControlProp extends Control = Control> = {
```
Comment (kotarella1110) [inline src/types.ts]: is this `ControlProps` generics parameter name OK?

File: src/controller.tsx (hunk 2 above)
Comment (kotarella1110) [reply src/controller.tsx]: I will!

Comment (kotarella1110) [reply src/controller.tsx]: @bluebill1049 completed!

File: src/types.ts (hunk 3 above)
Comment (bluebill1049) [reply src/types.ts]: yeah

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**